app-modules/ai/src/Services/Contracts/AiService.php

```diff
@@ -56,6 +57,11 @@ public function createAssistant(AiAssistant $assistant): void;
      */
     public function updateAssistant(AiAssistant $assistant): void;
 
+    /**
+     * Override and update the existing tools set on an Assistant
+     */
+    public function updateAssistantTools(AiAssistant $assistant, array $tools): void;
```

Comment:
I am unsure if the concept of "tools" should leak outside of the Open AI module. How about something more specific for this feature such as `enableAssistantFileUploads()` as part of the `SupportsFileUploads` contract? And then checking for that contract before enabling file uploads.